app/src/main/java/com/eis/geoCalendar/app/GenericEventManager.java

```diff
@@ -0,0 +1,97 @@
+package com.eis.geoCalendar.app;
+
+import androidx.annotation.NonNull;
+
+import com.eis.geoCalendar.events.Event;
+import com.eis.geoCalendar.events.EventDatabase;
+import com.eis.geoCalendar.events.EventManager;
+import com.eis.geoCalendar.gps.GPSPosition;
+
+import java.util.ArrayList;
+
+/**
+ * This class manages all generic events.
+ *
+ * @param <E> The generic timed event.
+ * @author Francesco Bau'
+ * @version 1.0
+ * @see com.eis.geoCalendar.events.Event
+ * @see com.eis.geoCalendar.events.EventManager
+ * @since 24/12/2019
+ */
+public class GenericEventManager<E extends Event> implements EventManager<E> {
+
+    protected EventDatabase<E> database;
+
+    /**
+     * Main constructor, needed to acquire the database of the events.
+     *
+     * @param database The database of the events.
+     */
+    public GenericEventManager(EventDatabase<E> database) {
+        this.database = database;
+    }
+
+    /**
+     * Adds an event to the local event list.
+     *
+     * @param event The event to add.
+     */
+    @Override
+    public void addEvent(@NonNull E event) {
+        database.saveEvent(event);
+    }
+
+    /**
+     * Removes an event from the local event list.
+     *
+     * @param event The event to remove.
+     * @return {@code true} if the event was present and has been removed, {@code false} otherwise.
+     */
+    @Override
+    public boolean removeEvent(@NonNull E event) {
+        return database.removeEvent(event);
+    }
+
+    /**
+     * Searches for events in a given circle of gps positions.
+     *
+     * @param p     The center of the circle of search.
+     * @param range The radius of the circe of search in meters.
+     * @return {@link ArrayList} of events in the given area. It's empty if there is none.
+     */
+    @Override
+    public ArrayList<E> getEventsInRange(@NonNull GPSPosition p, float range) {
+        ArrayList<E> eventsList = new ArrayList<>();
+        for (E event : database.getSavedEvents()) {
+            if (event.getPosition().getDistance(p) < range)
+                eventsList.add(event);
+        }
+        return eventsList;
+    }
+
+    /**
+     * Searches for the closest event to a given {@link GPSPosition}
+     *
+     * @param p The position used to calculate the closest event
+     * @return The closest event to the given Position if there is one, {@code null} otherwise
+     */
+    @Override
+    public E getClosestEvent(@NonNull GPSPosition p) {
+        if (database.getSavedEvents().isEmpty()) return null;
+        E relativeClosestEvent = database.getSavedEvents().get(0);
```

Comment:
I think that that 0 it's hardwiring, i'd create an int object named firstEvent or comparisonEvent.
private final int comparisonEvent=0;

Reply:
You might be right, but 0 is obvious to mean "the first element".
I've seen the other code I have to review, it has plenty of magic numbers, even if the coders are way more experts than me...
So I don't know if that 0 is considered hardwiring...)

Reply:
You might be right, but 0 is obvious to mean "the first element".
I've seen the other code I have to review, it has plenty of magic numbers, also zeros meaning "the first element".... even if the coders are way more experts than me. So I ignored those zeros...
So I don't know if that 0 is considered hardwiring...
In doubt, I'm gonna fix it now, though... Since I have only a few zeros in these classes.
Thank you for pointing this out.

Reply:
I talked with experts, and they said it's not hardwiring, because 0, in this case, means "the first element", and it would never happen that, in the future, an ArrayList's first element would change.
Putting a constant, in this case, would make the code more fragile, while a constant's meaning is to edit a specific thing that might be in more classes.
Am I right, @Cogno-IDU ?

Reply:
That's how I would consider it, yeah.
Try one thing though, if it exist `get(0)` then there might be a function like `first()` or `pop()` that does the same thing, which might be what you're looking for.

Reply:
Unfortunately, since it's an ArrayList, there's no such thing.
To have these kinds of methods, I should have used a class which extends ArrayList and implements a Queue or a Stack.... Right?

Reply:
no, I thought there already was a method doing that for you, but seems like there's not, so ¯\_(ツ)_/¯

Reply:
@kummy98 This is not considered hardwiring as the first element is always `get(0)` and that can't be changed. This `get(0)` doesn't represent the closest event all the time, it represents the "closest event until I know until this moment with this index" and when the index gets to the `size()` value it's actually the closest.